secretmgmt/secret.go

```diff
@@ -1,35 +1,35 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-package graph
+package secretmgmt
 
 import (
 	"github.com/Azure/acr-builder/util"
 	"github.com/pkg/errors"
 )
 
 var (
-	errMissingSecretID       = errors.New("secret is missing an ID")
-	errMissingSecretProps    = errors.New("secret is missing required akv property")
+	errMissingSecretIDs      = errors.New("secret is missing an ID as well as auto-generated ID")
+	errMissingSecretProps    = errors.New("secret should contain either akv property for vault secret, or msi clientID/armResourceId for msi authentication")
 	errSecretIDContainsSpace = errors.New("secret ID cannot contain spaces")
 	errInvalidUUID           = errors.New("msi client ID is not a valid guid")
 )
 
-// SecretValue represents the resolved value of the Secret.
-type SecretValue struct {
-	ID    string
-	Value string
-}
-
 // Secret defines a wrapper to resolve vault secrets to values.
 type Secret struct {
 	ID          string `yaml:"id"`
 	Akv         string `yaml:"akv,omitempty"`
 	MsiClientID string `yaml:"clientID,omitempty"`
 
+	// After the Secret is resolved, the value can be found here.
+	ResolvedValue string
+
+	// ArmResourceID is used to fetch ARM token from a TokenServer for an identity
+	ArmResourceID string
```

Comment:
Should we add this ArmResourceID to include MSI secrets here? Just how Akv exists as a part of "Secret"
Or should we split this into two classes: `MSISecret` and `VaultSecret`.
Since when it comes to MSI secrets `ARMResourceID`, we have full control over the `ID` property here. Since it is not entered by user. Right now, I use it to send `RegistryName` so we can obtain a token for the registry. What I could do is make the `ID` "registryName_armResourceId` and delete this field. 
@northtyphoon  @ehotinger